tests/pylint/censorship.py

```diff
@@ -0,0 +1,188 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+# Taken from:
+# https://github.com/jkonecny12/pylint_censorship
+
+__all__ = ["CensoreshipLinter", "CensoreshipConfig"]
+
+import sys
+import re
+import pylint.lint
+
+from io import StringIO
+
+from pylint.reporters.text import TextReporter
+
+
+class FalsePositive(object):
+    """An object used in filtering out incorrect results from pylint.  Pass in
+       a regular expression matching a pylint error message that should be
+       ignored.  This object can also be used to keep track of how often it is
+       used, for auditing that false positives are still useful.
+    """
+    def __init__(self, regex):
+        self.regex = regex
+        self.used = 0
+
+
+class CensoreshipConfig(object):
+    """Configuration of False Positives you want to run by Pylint."""
+    def __init__(self):
+        """Create a configuration object.
+
+        Attributes:
+
+        false_possitives: List of FalsePositive instances you want to filter out.
+
+        pylintrc_path: Path to the Pylint configuration file. Everything except false positives
+                       should be configured there. You can also pass pylintrc as argument to
+                       command_line_args in that case the command_line_args rc file will
+                       taken instead.
+
+        command_line_args: Pass this list of command_line_args to pylint.
+        """
+        self.false_positives = []
+        self.pylintrc_path = ""
+        self.command_line_args = []
+
+    @property
+    def check_paths(self):
+        """Get paths to check.
+
+        These can be python modules or files.
+
+        :return: string with paths separated by space
+        """
+        raise AttributeError("No test paths are specified. Please override "
+                             "CensoreshipConfig.check_paths property!")
+
+
+class CensoreshipLinter(object):
+    """Run pylint linter and modify it's output."""
+
+    def __init__(self, config):
+        """Create CenshoreshipLinter class.
+
+        :param config: configuration class for this Linter
+        :type config: CensoreshipConfig class instance
+        """
+        self._stdout = StringIO()
+        self._config = config
+
+    def run(self):
+        """Run the pylint static linter.
+
+        :return: return code of the linter run
+        :rtype: int
+        """
+        args = self._prepare_args()
+
+        print("Running pylint with args: ", args)
+
+        pylint.lint.Run(args,
```

Comment:
I thought that pylint will run separately now. We need to know only the pylint output to handle the false positives. I don't mind to wrap pylint in another class, but it shouldn't be this one.

Reply:
We can talk about possible solution. Right now I wanted to be as much as possible close to the pocketlint implementation.

Reply:
Ok, it can stay as it is.